internal/encryption/keyprovider/gcp_kms/config.go

```diff
@@ -0,0 +1,149 @@
+package gcp_kms
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/opentofu/opentofu/internal/encryption/keyprovider"
+	"github.com/opentofu/opentofu/internal/httpclient"
+	"github.com/opentofu/opentofu/version"
+	"golang.org/x/oauth2"
+	"google.golang.org/api/impersonate"
+	"google.golang.org/api/option"
+
+	kms "cloud.google.com/go/kms/apiv1"
+)
+
+type keyManagementClientInit func(ctx context.Context, opts ...option.ClientOption) (keyManagementClient, error)
+
+// Can be overridden for test mocking
+var newKeyManagementClient keyManagementClientInit = func(ctx context.Context, opts ...option.ClientOption) (keyManagementClient, error) {
+	return kms.NewKeyManagementClient(ctx, opts...)
+}
+
+type Config struct {
+	Credentials string `hcl:"credentials,optional"`
+	AccessToken string `hcl:"access_token,optional"`
+
+	ImpersonateServiceAccount          string   `hcl:"impersonate_service_account,optional"`
+	ImpersonateServiceAccountDelegates []string `hcl:"impersonate_service_account_delegates,optional"`
+
+	KMSKeyName string `hcl:"kms_encryption_key"`
+	KeyLength  int    `hcl:"key_length"`
+}
+
+func stringAttrEnvFallback(val string, env string) string {
+	if val != "" {
+		return val
+	}
+	return os.Getenv(env)
+}
+
+// TODO This is copied in from the backend packge to prevent a circular dependency loop
+// If the argument is a path, Read loads it and returns the contents,
+// otherwise the argument is assumed to be the desired contents and is simply
+// returned.
```

Comment:
I'm a bit confused, I don't see something called `Read` here

Reply:
That's an issue in the original function that's in the backend package.  It was renamed a long time ago and the documentation was not updated.

Reply:
Could we update/remove it for this PR? Im not a big fan of bringing across things and not cleaning them up.

Reply:
Fixed it in both places.